test.sh

```diff
@@ -1,80 +1,132 @@
+# 5. test.sh should validate i have the AWS keys, and a CLUSTER_URL set, but it need not verify the azure keys / security / etc
 
-# Exit immediately on errors -- the helper scripts all emit github statuses internally
+# Exit immediately on errors
 set -e
 
-function run_framework_tests {
-    framework=$1
-    FRAMEWORK_DIR=${REPO_ROOT_DIR}/frameworks/${framework}
+REPO_ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+FRAMEWORK_LIST=$(ls $REPO_ROOT_DIR/frameworks | sort)
 
-    # Build/upload framework scheduler artifact if one is not directly provided:
-    if [ -z "${!STUB_UNIVERSE_URL}" ]; then
-        STUB_UNIVERSE_URL=$(echo "${framework}_STUB_UNIVERSE_URL" | awk '{print toupper($0)}')
-        # Build/upload framework scheduler:
-        UNIVERSE_URL_PATH=$FRAMEWORK_DIR/${framework}-universe-url
-        UNIVERSE_URL_PATH=$UNIVERSE_URL_PATH ${FRAMEWORK_DIR}/build.sh aws
+function usage()
+{
+    echo "Usage: $0 [-m MARKEXPR] [-k EXPRESSION] [-p PATH] [-s] all|<framework-name>"
+    echo "-m passed to pytest directly [default -m \"sanity and not azure\"]"
+    echo "-k passed to pytest directly [default NONE]"
+    echo "-p PATH to cluster SSH key [default ~/.ssh/ccm.pem]"
+    echo "-s run in strict mode (sets \$SECURITY=\"strict\")"
+    echo "Cluster must be created and \$CLUSTER_URL set"
+    echo "AWS credentials must exist in the variables:"
+    echo "      \$AWS_ACCESS_KEY_ID"
+    echo "      \$AWS_SECRET_ACCESS_KEY"
+    echo "Azure tests will run if thses variables are set:"
+    echo "      \$AWS_ACCESS_KEY_ID"
+    echo "      \$AWS_SECRET_ACCESS_KEY"
+    echo "      \$AZURE_CLIENT_ID"
+    echo "      \$AZURE_CLIENT_SECRET"
+    echo "      \$AZURE_TENANT_ID"
+    echo "      \$AZURE_STORAGE_ACCOUNT"
+    echo "      \$AZURE_STORAGE_KEY"
+    echo "  (changes the -m default to \"sanity\")"
+    echo "Set \$STUB_UNIVERSE_URL to bypass build"
+    echo "  (invalid when building all frameworks)"
+    echo "Current frameworks:"
+    for framework in $FRAMEWORK_LIST; do
+        echo "       $framework"
+    done
+    exit 1
+}
 
-        if [ ! -f "$UNIVERSE_URL_PATH" ]; then
-            echo "Missing universe URL file: $UNIVERSE_URL_PATH"
-            exit 1
-        fi
-        export STUB_UNIVERSE_URL=$(cat $UNIVERSE_URL_PATH)
-        rm -f $UNIVERSE_URL_PATH
-        echo "Built/uploaded stub universe: $STUB_UNIVERSE_URL"
-    else
-        echo "Using provided STUB_UNIVERSE_URL: $STUB_UNIVERSE_URL"
-    fi
+if [ "$#" -eq "0" ]; then
+    usage
+fi
 
-    echo Security: $SECURITY
-    if [ "$SECURITY" = "strict" ]; then
-        ${REPO_ROOT_DIR}/tools/setup_permissions.sh root ${framework}-role
-        # include foldered roles (tests exercise with /test/integration/svcname):
-        ${REPO_ROOT_DIR}/tools/setup_permissions.sh root test__integration__${framework}-role
-    fi
+if [ -z "$CLUSTER_URL" ]; then
+    echo "Cluster not found. Create and configure one then set \$CLUSTER_URL."
+    exit 1
```

Comment:
Really?  We don't want to be able to have the cluster brought up the same way as CI does for reproding CI stuff?